Blog/gst-build-cross-build/cross-building-with-gstreamer.md

```diff
@@ -0,0 +1,320 @@
+
+```
+# ./generate-cross-file.py
+```
+
+```
+[host_machine]
+system = 'linux'
+cpu_family = 'aarch64'
+cpu = 'arm64'
+endian = 'little'
+
+[properties]
+c_args = ['-Wall', '-g', '-O2']
+cpp_args = ['-Wall', '-g', '-O2']
+objc_args = ['-Wall', '-g', '-O2']
+objcpp_args = []
+c_link_args = []
+cpp_link_args = []
+objc_link_args = []
+objcpp_link_args = []
+pkg_config_libdir = ['__unknown_sysroot__']
+
+
+[binaries]
+c = ['aarch64-linux-gnu-gcc']
+cpp = ['aarch64-linux-gnu-g++']
+ar = ['aarch64-linux-gnu-ar']
+pkgconfig = 'pkg-config'
+cmake = ['false']
+strip = ['aarch64-linux-gnu-strip']
+
+```
+
+Here `meson` will use `aarch64-linux-gnu-gxx` to compile with the given arguments setup above. As `meson` does not recommend to use environment variables, the cross file contains hard-coded path to the sysroot to provide package config.
+Indeed since `meson` > 0.54, you can define `pkg_config_libdir` which will help pkg-config to search for the package configuration files for the given target. You can also tell the path to the pkg-config wrapper by modifying the pkgconfig variable as well.
+Predefined cross file can also be found in `gst-build/data/cross-files`
+
+
+#### Configuring the project for Zynq UltraScale+ MPSoC rev F
+
+When the cross file is ready, we can now configure `gst-build` in order to have a dedicated build for our platform. Here I'm disabling some unnecessary options of `gst-build` such as *libav*, *vaapi* or *gtk_doc*.
+
+Please ensure that you have the last meson version, otherwise gst-build will take glib from the system.
+
+```
+# /path/to/meson_0_54 build-cross-arm64 --cross-file=my-meson-cross-file.txt -D omx=enabled -D sharp=disabled -D gst-omx:header_path=/opt/allegro-vcu-omx-il/omx_header -D gst-omx:target=zynqultrascaleplus -D libav=disabled -D rtsp_server=disabled -D vaapi=disabled -D disable_gst_omx=false -Dugly=disabled -Dgtk_doc=disabled
+
+```
+
+After this step, you should be able to build with `ninja`.
+
+```
+# ninja -C build-cross-arm64
+```
+
+#### Installing
+
+Last but not the least, you need to install the artifacts in a given folder to be mounted by your target with NFS by example. You have to provide a **DESTDIR** variable to `ninja` and it will install in `$DESTDIR/usr/local/` as install prefix.
+
+```
+# DESTDIR=/opt/gst-build-cross-artifacts ninja install
+```
+
+
+#### Running the binaries on target
+
+
+After mounting the folder or copying it to your target, you have to set up a few variables to be able to run GStreamer pipelines:
+
+ * PATH=$DESTDIR/usr/local/bin:$PATH
+ * LD_LIBRARY_PATH=$DESTDIR/usr/local/lib:$LD_LIBRARY_PATH
+ * GST_PLUGIN_PATH=$DESTDIR/usr/local/lib/gstreamer-1.0
+ * GST_OMX_CONFIG_DIR=$DESTDIR/usr/local/etc/xdg
+
+A python script is also available [here](https://github.com/dabrain34/gstreamer-toolkit/blob/master/gst-build-helper/cross-gst-uninstalled.py) to setup the correct environment for your environment.
```

Comment:
Can't we copy the whole build dir instead and rely on `meson uninstalled` to set the env variables?

Reply:
it was to avoid the python dependencies which is stupid because my script in written in python, I should have it in bash script :)